harness/determined/core/_train.py

```diff
@@ -88,15 +88,12 @@ def _report_trial_metrics(
             serializable_metrics = self._get_serializable_metrics(metrics)
             reportable_metrics = {k: metrics[k] for k in serializable_metrics}
 
-        v1metrics = bindings.v1Metrics(avgMetrics=reportable_metrics, batchMetrics=batch_metrics)
-        v1TrialMetrics = bindings.v1TrialMetrics(
-            metrics=v1metrics,
-            stepsCompleted=steps_completed,
-            trialId=self._trial_id,
-            trialRunId=self._run_id,
+        self._metrics.report(
```

Comment:
general question: do we have sufficient automated testing for this.
1. do we have an existing unit test, or is it possible to have a unit test testing completeness/correctness of metrics sent from the harness? 
2. if not possible: do we have an e2e test which'd check these?

Reply:
good question. the full end-to-end journey of metrics probably goes something like:
training code -> trial APIs -> core API (now with async logic)-> POST APIs to master -> master side stuff -> read APIs

individual trials have unit tests (`test_tf_keras_trial.py`, `test_pytorch_trial.py`) that test the Trial API generates the expected metrics (training code -> trial APIs) but we don't have unit tests for the core API -> master APIs leg of metrics reporting. i'll write one. 

beyond that the end-to-end training code -> metrics read APIs is covered by `e2e_tests/.../test_metrics.py`

Reply:
added unit tests for the new `MetricsContext`